mpd/asyncio.py

```diff
@@ -0,0 +1,406 @@
+#                elif key in obj:
+#                    if not isinstance(obj[key], list):
+#                        obj[key] = [obj[key], value]
+#                    else:
+#                        obj[key].append(value)
+#                    continue
+#            obj[key] = value
+#        if obj:
+#            yield obj
+
+    def _parse_objects_direct(self, lines, delimiters=[]):
+        # This is a workaround implementing the above comment on Python 3.5. It
+        # is recommended that the commented-out code be used for reasoning, and
+        # that changes are applied there and only copied over to this
+        # implementation.
+
+        outerself = self
+        class WrappedLoop:
+            def __init__(self):
+                self.obj = {}
+                self.exhausted = False
+
+            def __aiter__(self):
+                return self
+
+            async def __anext__(self):
+                while True:
+                    if self.exhausted:
+                        raise StopAsyncIteration()
+
+                    line = await lines.get()
+                    if isinstance(line, BaseException):
+                        raise line
+                    if line is None:
+                        self.exhausted = True
+                        if self.obj:
+                            return self.obj
+                        continue
+                    key, value = outerself._parse_pair(line, separator=": ")
+                    key = key.lower()
+                    if self.obj:
+                        if key in delimiters:
+                            oldobj = self.obj
+                            self.obj = {key: value}
+                            return oldobj
+                        elif key in self.obj:
+                            if not isinstance(self.obj[key], list):
+                                self.obj[key] = [self.obj[key], value]
+                            else:
+                                self.obj[key].append(value)
+                            continue
+                    self.obj[key] = value
+        return WrappedLoop()
+
+    # command provider interface
+
+    @classmethod
+    def add_command(cls, name, callback):
+        command_class = CommandResultIterable if callback.mpd_commands_direct else CommandResult
+        if hasattr(cls, name):
+            # Idle and noidle are explicitly implemented, skipping them.
+            return
+        def f(self, *args):
+            result = command_class(name, args, partial(callback, self))
+            if self.__run_task is None:
+                raise ConnectionError("Can not send command to disconnected client")
+            self.__commandqueue.put_nowait(result)
+            self._nudge_idle()
+            return result
+        escaped_name = name.replace(" ", "_")
+        f.__name__ = escaped_name
+        setattr(cls, escaped_name, f)
+
+    # commands that just work differently
+
+#    async def idle(self, subsystems=()):
```

Comment:
What's up with this version of ``idle``? Can it be removed?

Reply:
same